### PR TITLE
[VBC] Select multiple legends for Vertical bar chart

### PR DESCRIPTION
**Pull request by srmukher**
Select multiple legends for Vertical bar chart

image

### CODE REVIEW

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ export interface IVerticalBarChartState extends IBasestate {
   hoverXValue?: string | number | null;
```
Comment (fabricteam) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:72]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33510&commitId=ed027b225ae82f5213bcdb883fb8e287324d4361&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 5 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Bottom auto edge.chromium.png|2309| Changed|
|Callout.Left top edge.chromium.png|1949| Changed|
|Callout.No beak.chromium.png|2306| Changed|
|Callout.Top center.chromium.png|2178| Changed|
|Callout.Top left edge.chromium.png|2307| Changed|

</details>


  <details><summary><b>Keytip</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Keytip.Offset.chromium.png|121| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.chromium.png|11| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|
|react-charting-LineChart.Events.chromium.png|2| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 2]
```diff
@@ -118,8 +119,8 @@ export class VerticalBarChartBase
       dataForHoverCard: 0,
       isCalloutVisible: false,
       refSelected: null,
-      selectedLegend: props.legendProps?.selectedLegend ?? '',
-      activeLegend: '',
+      selectedLegends: [],
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:122]: >[] [](http://example.com/codeflow?start=23&length=2)

set initial value as selectedLegend

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 3]
```diff
@@ -531,7 +533,7 @@ export class VerticalBarChartBase
       : this._createColors()(1);
 
     // there might be no y value of the line for the hovered bar. so we need to check this condition
-    if (this._isHavingLine && selectedPoint[0].lineData?.y !== undefined) {
+    if (this._isHavingLine && selectedPoint[0].lineData?.y !== undefined && this._legendHighlighted(lineLegendText)) {
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:536]: >this._legendHighlighted(lineLegendText) [](http://example.com/codeflow?start=76&length=39)

this condition should be true if no legend highlighted as well

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 4]
```diff
@@ -541,18 +543,20 @@ export class VerticalBarChartBase
...
-      y: selectedPoint[0].y,
-      color: enableGradient
-        ? useSingleColor
-          ? getNextGradient(0, 0)[0]
-          : selectedPoint[0].gradient?.[0] || getNextGradient(pointIndex, 0)[0]
-        : calloutColor,
-      data: selectedPoint[0].yAxisCalloutData,
-      yAxisCalloutData: selectedPoint[0].yAxisCalloutData,
-    });
+    if (this._legendHighlighted(selectedPoint[0].legend)) {
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:546]: >this._legendHighlighted(selectedPoint[0].legend [](http://example.com/codeflow?start=8&length=47)

same here. This should also be enabled if no legend is highlighted

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx [hunk 5]
```diff
@@ -652,6 +652,84 @@ describe('Vertical bar chart - Subcomponent Legends', () => {
...
+      const bars = getById(container, /_VBC_bar/i);
+      const line = getById(container, /_VBC_line/i)[0];
+      const legends = screen.getAllByText((content, element) => element!.tagName.toLowerCase() === 'button');
+      expect(line).toBeDefined();
+      expect(bars).toHaveLength(8);
+      expect(legends).toHaveLength(9);
+      fireEvent.click(screen.getByText('just line'));
+      fireEvent.click(screen.getByText('Oranges'));
+      expect(line.getAttribute('opacity')).toEqual('1');
+      expect(screen.getByText('Dogs')).toHaveStyle('opacity: 0.67');
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx:670]: >toHaveStyle [](http://example.com/codeflow?start=39&length=11)

you are not checking for oranges opacity

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 3 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:536]: done

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 2 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:122]: done

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 4 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:546]: done

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx (hunk 5 above)
Comment (srmukher) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx:670]: added

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 2 above)
Comment (AtishayMsft) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:122]: I don't see it in the code.

Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:122]: >selectedLegends [](http://example.com/codeflow?start=6&length=15)

Could you take the change to set selectedLegends to props in componentDidUpdate as in this commit.
[https://github.com/microsoft/fluentui/commit/6a70a11ddb8bac592c4c9b4c12a2705a9ace6686](https://github.com/microsoft/fluentui/commit/6a70a11ddb8bac592c4c9b4c12a2705a9ace6686)

Also update the legendprops to enablemultiselect and selectedlegends value in declarative chart.tsx.

Same for VSBC and GVBC.

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 6]
```diff
@@ -118,8 +120,8 @@ export class VerticalBarChartBase
       dataForHoverCard: 0,
       isCalloutVisible: false,
       refSelected: null,
-      selectedLegend: props.legendProps?.selectedLegend ?? '',
-      activeLegend: '',
+      selectedLegends: props.legendProps?.selectedLegends || [],
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:123]: enable multiselect in DeclarativeChart.tsx also

Comment (srmukher) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:123]: done

File: packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx [hunk 7]
```diff
@@ -265,7 +265,7 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
       return (
         <VerticalBarChart
           {...transformPlotlyJsonToVBCProps(plotlySchema, colorMap, isDarkTheme)}
-          legendProps={legendProps}
+          legendProps={{ ...legendProps, canSelectMultipleLegends: true, selectedLegends: activeLegends }}
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:268]: Multiple charts are now using multiselect props. Create this as separate variable multiSelectLegendProps and replace for all charts where it is enabled.

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**